<launch>

  <arg name="rviz" default="true" />
  <arg name="scanPeriod" default="0.02" />

  <group ns="/ig/loam" >
      <rosparam command="load" file="$(find loam_velodyne)/config/ig_loam_config.yaml" />

      <node pkg="loam_velodyne" type="multiScanRegistration" name="multiScanRegistration" output="screen">
        <param name="lidar" value="VLP-16" /> <!-- options: VLP-16  HDL-32  HDL-64E -->
        <param name="scanPeriod" value="$(arg scanPeriod)" />
      </node>

      <node pkg="loam_velodyne" type="laserOdometry" name="laserOdometry" output="screen" respawn="true">
        <param name="scanPeriod" value="$(arg scanPeriod)" />
      </node>

      <node pkg="loam_velodyne" type="laserMapping" name="laserMapping" output="screen">
        <param name="scanPeriod" value="$(arg scanPeriod)" />
      </node>

      <node pkg="loam_velodyne" type="transformMaintenance" name="transformMaintenance" output="screen">
      </node>
  </group>

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find loam_velodyne)/rviz_cfg/loam_velodyne.rviz" />
  </group>

</launch>
